spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -122,8 +122,8 @@ private Spark3Util.CatalogAndIdentifier catalogAndIdentifier(CaseInsensitiveStri
     setupDefaultSparkCatalogs(spark);
     String path = options.get(SparkReadOptions.PATH);
 
-    Long snapshotId = propertyAsLong(options, SparkReadOptions.SNAPSHOT_ID);
-    Long asOfTimestamp = propertyAsLong(options, SparkReadOptions.AS_OF_TIMESTAMP);
+    Long snapshotId = PropertyUtil.propertyAsLong(options, SparkReadOptions.SNAPSHOT_ID, -1);
+    Long asOfTimestamp = PropertyUtil.propertyAsLong(options, SparkReadOptions.AS_OF_TIMESTAMP, -1);
```

Comment:
In this case you have to change the following code right to expect -1?

Reply:
Thanks for your help, have updated.